packages/math/src/index.ts

```diff
@@ -1,2 +1,261 @@
+  poolPairData: MetaStablePoolPairData
+) {
+  const spotPriceAfterSwapToStableMath =
+    MetaStableConversions.priceToStableMath(
+      spotPriceAfterSwap,
+      poolPairData.rateIn,
+      poolPairData.rateOut
+    );
+  const amountInToStableMath =
+    ExtendedStableMath._tokenInForExactSpotPriceAfterSwap(
+      spotPriceAfterSwapToStableMath,
+      poolPairData
+    );
+  return MetaStableConversions.amountFromStableMath(
+    amountInToStableMath,
+    poolPairData.rateIn
+  );
+}
+
+function tokenOutForExactSpotPriceAfterSwap(
+  spotPriceAfterSwap: OldBigNumber,
+  poolPairData: MetaStablePoolPairData
+) {
+  const spotPriceAfterSwapToStableMath =
+    MetaStableConversions.priceToStableMath(
+      spotPriceAfterSwap,
+      poolPairData.rateIn,
+      poolPairData.rateOut
+    );
+  const amountOutToStableMath =
+    ExtendedStableMath._tokenOutForExactSpotPriceAfterSwap(
+      spotPriceAfterSwapToStableMath,
+      poolPairData
+    );
+  return MetaStableConversions.amountFromStableMath(
+    amountOutToStableMath,
+    poolPairData.rateOut
+  );
+}
+
+function effectivePriceForExactTokenInSwap(
+  amountIn: OldBigNumber,
+  poolPairData: MetaStablePoolPairData
+): OldBigNumber {
+  const amountInToStableMath = MetaStableConversions.amountToStableMath(
+    amountIn,
+    poolPairData.rateIn
+  );
+  const effectivePriceToStableMath =
+    ExtendedStableMath._effectivePriceForExactTokenInSwap(
+      amountInToStableMath,
+      poolPairData
+    );
+  return MetaStableConversions.priceFromStableMath(
+    effectivePriceToStableMath,
+    poolPairData.rateIn,
+    poolPairData.rateOut
+  );
+}
+
+function effectivePriceForExactTokenOutSwap(
+  amountOut: OldBigNumber,
+  poolPairData: MetaStablePoolPairData
+): OldBigNumber {
+  const amountOutToStableMath = MetaStableConversions.amountToStableMath(
+    amountOut,
+    poolPairData.rateOut
+  );
+  const effectivePriceToStableMath =
+    ExtendedStableMath._effectivePriceForExactTokenOutSwap(
+      amountOutToStableMath,
+      poolPairData
+    );
+  return MetaStableConversions.priceFromStableMath(
+    effectivePriceToStableMath,
+    poolPairData.rateIn,
+    poolPairData.rateOut
+  );
+}
+
+function priceImpactForExactTokenInSwap(
+  amountIn: OldBigNumber,
+  poolPairData: MetaStablePoolPairData
+): OldBigNumber {
+  const effectivePriceMetastable = effectivePriceForExactTokenInSwap(
+    amountIn,
+    poolPairData
+  );
+  const spotPriceMetastable = spotPrice(poolPairData);
+  return bnum(1).minus(effectivePriceMetastable.div(spotPriceMetastable));
+}
+
+function priceImpactForExactTokenOutReversedSwap(
+  amountOut: OldBigNumber,
+  poolPairData: MetaStablePoolPairData
+): OldBigNumber {
+  const effectivePriceMetastable = effectivePriceForExactTokenOutSwap(
+    amountOut,
+    poolPairData
+  );
+  const spotPriceMetastable = spotPrice(poolPairData);
+  return bnum(1).minus(spotPriceMetastable.div(effectivePriceMetastable));
+}
+
+export const MetaStableMath = {
```

Comment:
`#note` we should probably rename this package to something else 😝

Reply:
kkkkkk ok